src/shared/settings/danger-zone.tsx

```diff
@@ -0,0 +1,193 @@
+import { FormContainer } from "../onefx-auth-provider/email-password-identity-provider/view/form-container";
+import { InputError } from "../onefx-auth-provider/email-password-identity-provider/view/input-error";
+import { InputLabel } from "../onefx-auth-provider/email-password-identity-provider/view/input-label";
+import { TextInput } from "../onefx-auth-provider/email-password-identity-provider/view/text-input";
+
+const { Text } = Typography;
+const VERIFICATION_FORM = "VERIFICATION_FORM";
+
+const DELETE_ACCOUNT = gql`
+  mutation deleteAccount($email: String!) {
+    deleteAccount(deleteAccountInput: { email: $email })
+  }
+`;
+
+function validateEmail(email: string): boolean {
+  const re = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+  return re.test(String(email).toLowerCase());
+}
+
+type State = {
+  modalShow: boolean;
+  emailError: string;
+  codeError: string;
+};
+
+export class DangerZone extends Component<{}, State> {
+  public constructor(props: any) {
+    super(props);
+    this.state = {
+      modalShow: false,
+      emailError: "",
+      codeError: ""
+    };
+  }
+
+  public showModal = () => {
+    this.setState({
+      modalShow: true
+    });
+  };
+
+  public handleOk = async (func: MutationFn<{ email: string }>) => {
+    const el = window.document.getElementById(
+      VERIFICATION_FORM
+    ) as HTMLFormElement;
+    const { email = "", code = "" } = serialize(el, {
+      hash: true
+    }) as {
+      email: string;
+      code: string;
+    };
+    window.console.log(email);
+
+    if (code !== t("settings.verification_code")) {
+      notification.error({
+        message: t("settings.forbidden_action"),
+        description: t("settings.invalid_code"),
+        duration: 3
+      });
+      return;
+    }
+    if (!validateEmail(email)) {
+      notification.error({
+        message: t("settings.forbidden_action"),
+        description: t("auth/invalid-email"),
+        duration: 3
+      });
+      return;
+    }
+
+    const result: any = await func({
+      variables: { email }
+    });
+
+    this.setState({
+      modalShow: false
+    });
+
+    if (result.data.deleteAccount) {
+      window.location.href = "/logout";
+      window.console.log(result);
+    } else {
+      notification.error({
+        message: t("settings.forbidden_action"),
+        description: t("auth/invalid-email"),
+        duration: 3
+      });
+    }
+  };
+
+  public handleCancel = () => {
+    this.setState({
+      modalShow: false
+    });
+  };
+
+  public render(): JSX.Element {
+    const { emailError, codeError } = this.state;
+
+    return (
+      <Flex
+        flexDirection="row"
+        alignContent="center"
+        justifyContent="space-between"
+      >
+        <Flex flexDirection="column" justifyContent="center" alignItems="start">
+          <Text style={{ fontSize: 20, fontWeight: "bold" }}>
```

Comment:
<img width="484" alt="Screen Shot 2019-11-05 at 11 47 15 PM" src="https://user-images.githubusercontent.com/1387285/68278637-a37f2f00-0026-11ea-9935-2bda45e51e38.png">

need some margin between the text and the button